src/mlpack/tests/main_tests/krann_test.cpp

```diff
@@ -0,0 +1,553 @@
+/**
+ * @file krann_test.cpp
+ * @author Ryan Curtin
+ * @author Utkarsh Rai
+ *
+ * Test mlpackMain() of krann_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "K-RankApproximateNearestNeighborsSearch";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/rann/krann_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct KRANNTestFixture
+{
+  KRANNTestFixture()
+  {
+    CLI::RestoreSettings(testName);
+  }
+
+  ~KRANNTestFixture()
+  {
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(KRANNMainTest, KRANNTestFixture);
+
+/*
+ * Check that we can't provide reference and query matrices
+ * with different dimensions.
+ */
+BOOST_AUTO_TEST_CASE(KRANNEqualDimensionTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(3, 100); // 100 points in 3 dimensions.
+
+  // Now we specify an invalid dimension(2) for the query data.
+  // Note that the number of points in queryData and referenceData matrices
+  // are allowed to be different
+  arma::mat queryData;
+  queryData.randu(2, 90); // 90 points in 2 dimensions.
+
+  // Random input, some k <= number of top tau %ile reference points.
```

Comment:
Do you mind to use `percentile` instead of `%ile` I guess for people where English is not the first language this can be confusions?